quinn-udp/src/unix.rs

```diff
@@ -830,6 +926,36 @@ mod gso {
         }
     }
 
+    fn supported_by_current_kernel() -> bool {
+        let kernel_version_string = match linux::kernel_version_string() {
+            Ok(kernel_version_string) => kernel_version_string,
+            Err(_errno) => {
+                crate::log::warn!(
+                    "Failed to retrieve kernel version string, GSO not enabled ({_errno})"
+                );
+                return false;
+            }
+        };
+
+        let kernel_version = match linux::KernelVersion::from_str(&kernel_version_string) {
+            Ok(kernel_version) => kernel_version,
+            Err(_reason) => {
+                crate::log::warn!("GSO not enabled: {}", _reason);
+                return false;
+            }
+        };
+
+        if kernel_version < SUPPORTED_SINCE {
+            crate::log::info!(
+                "GSO supported on Linux kernels 4.18+, current is {:?}",
+                kernel_version_string
+            );
+            false
```

Comment:
```suggestion
            return false
```

Then you can remove the `else`.